src/apps/vpn/ui/screens/devices/VPNDeviceList.qml

```diff
@@ -112,6 +112,13 @@ ListView {
         }
     }
 
+    Connections {
+        target: VPNDeviceModel
+        function onActiveDevicesChanged() {
+            if (!listView.anySwipesOpen()) listView.isEditing(false)
```

Comment:
Assume that a swipe is open for Device1 and Device1 was just removed by signing out from another device, causing this code to execute. Will Device1 remain in listView (until the screen is reloaded)? Then, if the user selects delete in the swipe, will that fail silently because the device has already been deleted?

Or is it better to automatically update listView when Device1 is removed?

Reply:
`listView`'s model is bound to `VPNDeviceModel` so anytime there is a change to `VPNDeviceModel`, `listView` automatically updates. If the model is changed by something outside the UI (i.e. signing out from another device), the list will refresh and all swipes will be closed.

Reply:
Thanks for the explanation.